modules/registry/registry-server/registry-api-service/src/main/java/org/apache/airavata/registry/api/service/handler/RegistryServerHandler.java

```diff
@@ -3589,8 +3586,7 @@ public String registerApplicationDeployment(String gatewayId, ApplicationDeploym
     @Override
     public boolean updateApplicationModule(String appModuleId, ApplicationModule applicationModule) throws RegistryServiceException, TException {
         try {
-            appCatalog = RegistryFactory.getAppCatalog();
-            appCatalog.getApplicationInterface().updateApplicationModule(appModuleId, applicationModule);
+            new ApplicationInterfaceRepository().updateApplicationModule(appModuleId, applicationModule);
```

Comment:
@machristie @tilaks26, Just for my knowledge, is this the new way to do database operations? Is it required to create a new object of ApplicationInterfaceRepository for each call? Can't we re use them?

Reply:
My understanding is that we can reuse them since they are stateless and hence threadsafe. +1 for just instantiating these once in RegistryServerHandler and reusing them.

Reply:
@machristie @DImuthuUpe Same thing has been followed in here in the SharingRegistryServerHandler https://git.io/vx0vX. Wonder what the reason could be

Reply:
@sachinkariyattin yeah, I don't know. I don't see any reason why they couldn't be instantiated just once since they have no state.